### PR TITLE
Display error message in Add Forcefield Dialog

### PR DESCRIPTION
**Pull request by rprospero**
This closes issue #884.  There's a screenshot on the issue that shows the current status of the messages in the GUI.

### CODE REVIEW

File: src/data/ff/ff.cpp [hunk 1]
```diff
@@ -304,15 +304,15 @@ int Forcefield::assignAtomTypes(Species *sp, CoreData &coreData, AtomTypeAssignm
...
+            failed.push_back(i.Z());
         }
     }
 
-    if (nFailed)
-        Messenger::error("Failed to assign atom {} to {} {}.\n", (nFailed == 1 ? "type" : "types"), nFailed,
-                         (nFailed == 1 ? "atom" : "atoms"));
+    if (!failed.empty())
+        Messenger::error("Failed to assign atom {} to {} {}.\n", (failed.size() == 1 ? "type" : "types"), failed.size(),
+                         (failed.size() == 1 ? "atom" : "atoms"));
```
Comment (trisyoungs) [inline src/data/ff/ff.cpp:313]: Gets a bit awkward here if one thinks about all of the possibilities.  For instance, some carbon atoms could be successfully assigned types, while others might not. We could hedge our bets and change the message to `Failed to assign atom types to (some)...`?

File: src/data/ff/ff.h [hunk 2]
```diff
@@ -185,7 +185,9 @@ class Forcefield
         SelectionOnlyFlag = 4      /* Only assign terms where all atoms are in the current selection */
     };
     // Assign suitable AtomTypes to the supplied Species, returning the number of failures
-    int assignAtomTypes(Species *sp, CoreData &coreData, AtomTypeAssignmentStrategy strategy, bool setSpeciesAtomCharges) const;
+    // Returns any elements that were unassigned
+    std::vector<Elements::Element> assignAtomTypes(Species *sp, CoreData &coreData, AtomTypeAssignmentStrategy strategy,
+                                                   bool setSpeciesAtomCharges) const;
```
Comment (trisyoungs) [inline src/data/ff/ff.h:190]: A possible future extension here could be to return the list of atom indices that failed to type correctly, and adding a new page into the wizard to display a viewer with those atoms highlighted in the model.  But definitely not for this PR!

File: src/gui/addforcefieldtermsdialog.h [hunk 3]
```diff
@@ -103,7 +103,8 @@ class AddForcefieldTermsDialog : public WizardDialog
     bool intramolecularTermsAssigned_;
     // Assign intramolecular terms to species
     bool assignIntramolecularTerms(const Forcefield *ff);
-
+    // Helper function for error messages
+    bool alertAboutAtomTypeErrors(std::vector<Elements::Element> errs);
```
Comment (trisyoungs) [inline src/gui/addforcefieldtermsdialog.h:107]:
```suggestion
    bool alertAboutAtomTypeErrors(std::vector<Elements::Element> errs);

```

**Review by trisyoungs — APPROVED** [review]
:+1: Just some small suggestions.

File: src/data/ff/ff.h (hunk 2 above)
Comment (trisyoungs) [reply src/data/ff/ff.h:190]: I guess this is sensible - sorry for dragging this PR out!  Don't forget to return the "user" indices (1:N) for the atoms so as not to confuse us chemists who aren't zero-indexed...

Comment (rprospero) [reply src/data/ff/ff.h]: I've moved over to the index based reporting

File: src/data/ff/ff.cpp (hunk 1 above)
Comment (rprospero) [reply src/data/ff/ff.cpp]: We now return the specific atomic indices, so this shouldn't be an issue.

**Pull request merged**